currency.js

```diff
@@ -70,10 +36,32 @@ const supportedCurrencies = [
     'ZAR'
 ];
 
-const number = '(-?[0-9](,|[0-9])*(.[0-9]+)?) ?';
-const currencies = `(${supportedCurrencies.join('|')})`;
-const regex = `${number} ?${currencies}( in ${currencies})?`;
-const re = new RegExp(regex, 'gi');
+const re = (() => {
+    const number = '(-?[0-9](,|[0-9])*(.[0-9]+)?) ?';
+
+    const aliases = [];
+    for (let key in currencyInfo.aliases) {
+        aliases.push(key);
+    }
+    const currencies = `(${supportedCurrencies.join('|')}|${aliases.join('|')})`;
+    const regex = `${number} ?${currencies}( in ${currencies})?`;
+    return new RegExp(regex, 'gi');
+})();
+
+const findCurrency = (cur) => {
+    cur = cur.toUpperCase();
+    // If cur is already a currency, return it.
+    if (currencyInfo.names[cur]) return cur;
+
+    // Otherwise, we need to find an alias:
+    for (var alias in currencyInfo.aliases) {
```

Comment:
`var` could be `let`.  
Would it not be better to store these as RegExp rather than recreate the RegExp every time?

Just for the sake of it, a one liner which I think would do this return.
```js
return currencyInfo.aliases[Object.keys(currencyInfo.aliases).find(a => cur.match(new RegExp(a, 'i')))] || null;
```